frontend/packages/dev-console/src/components/pipelines/pipeline-builder/types.ts

```diff
@@ -30,13 +30,34 @@ export type PipelineBuilderFormValues = PipelineBuilderTaskGrouping & {
   resources: PipelineResource[];
 };
 
-export type PipelineBuilderFormikValues = FormikValues & PipelineBuilderFormValues;
+export type PipelineBuilderFormYamlValues = {
+  editorType: EditorType;
+  yamlData: string;
+  formData: PipelineBuilderFormValues;
+};
+
+export type PipelineBuilderFormikValues = FormikValues & PipelineBuilderFormYamlValues;
 
 export type SelectedBuilderTask = {
   resource: PipelineResourceTask;
   taskIndex: number;
 };
 
+export enum UpdateOperationType {
+  ADD_LIST_TASK,
+  CONVERT_LIST_TO_TASK,
+  UPDATE_TASK,
+  REMOVE_TASK,
+  DELETE_LIST_TASK,
+  FIX_INVALID_LIST_TASK,
+}
+
+export enum TaskErrorType {
+  MISSING_REQUIRED_PARAMS = 'missingParams',
+  MISSING_NAME = 'nameMissing',
+  MISSING_RESOURCES = 'missingResources',
+}
```

Comment:
So types can have circular references. TypeScript handles this better than JS imports of code. You should be able to return these enums to the other file.

Reply:
The linter didn't let me commit, which is why I did the suffling. I've disabled the warnings now.

Reply:
Hmm, damn, that was not necessary. I didn't see the lint error on my side. I didn't try to commit naturally haha... and I didn't run the linter. Must be something my IDE is not picking up on. 🤦  This is worse actually then what you originally did... but I think we can leave it. I'll LGTM this PR and just get it merged. We can clean up once it's in it's own package.